### PR TITLE
ML-3654: Set result when using error handler in graph

### PR DESCRIPTION
**Pull request by davesh0812**
see also mlrun pr - https://github.com/mlrun/mlrun/pull/3341

### CODE REVIEW

File: storey/flow.py [hunk 1]
```diff
@@ -215,22 +215,30 @@ async def _do_and_recover(self, event):
...
-                else:
-                    raise ex
-            event.origin_state = self.name
-            event.error = ex
-            return await recovery_step._do(event)
+            message = traceback.format_exc()
+            if recovery_step:
+                event.origin_state = self.name
+                if event._awaitable_result:
+                    event._awaitable_result._set_result(Event(body={"error_msg": message, "origin_body": event.body}))
```
Comment (gtopper) [inline storey/flow.py:222]: You really can't just make up an event to put there. You're thinking about serving, but storey isn't just used for serving.

Comment (gtopper) [reply storey/flow.py:222]: Furthermore, there was already a call to `_set_error` there, which calls `_set_result`.

Comment (gtopper) [reply storey/flow.py:222]: It also makes no sense to set the result and then do the recovery. The recovery should lead somewhere that completes the result.

File: storey/flow.py [hunk 2]
```diff
@@ -215,22 +215,30 @@ async def _do_and_recover(self, event):
...
+                if self.logger:
+                    self.logger.error(
+                        f"step {self.name} got error {ex} when processing an event:\n {event.body}\n" f"{message}"
+                    )
+                await recovery_step._do(event)
+                return
```
Comment (gtopper) [inline storey/flow.py:228]: The result of `await recovery_step._do(event)` should be returned as before, not `None`. This matters for flow termination.

**Review by gtopper — CHANGES_REQUESTED** [review]
Please see my comments on [ML-3654](https://jira.iguazeng.com/browse/ML-3654) for RCA and proposed action points.

**Comment by davesh0812** [comment]
will redesign as discussed

**Pull request closed without merging**